src/typedefs.jl

```diff
@@ -17,6 +18,9 @@ mutable struct CategoricalPool{T, R <: Integer, V}
                                       invindex::Dict{T, R},
                                       order::Vector{R},
                                       ordered::Bool) where {T, R, V}
+        if iscatvalue(V) !== IsCatValue
```

Comment:
It would make sense to check that `reftype(V) === R`, else we could end up in a strange situation (even if in practice things might work).